pkg/controller/common/reconciler.go

```diff
@@ -198,11 +203,33 @@ func (r *BaseIngressReconciler) reconcileRoute(ctx context.Context, ci networkin
 }
 
 func (r *BaseIngressReconciler) reconcileDeletion(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
-	// TODO: something with a finalizer?  We're using owner refs for
-	// now, but really shouldn't be using owner refs from
-	// cluster-scoped ClusterIngress to a namespace-scoped K8s
-	// Service.
-	return nil
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	// get list of ingress object for a namespace
+	ingressList := networkingv1alpha1.IngressList{}
+	if err := r.Client.List(ctx, &client.ListOptions{
+		Namespace: ci.GetNamespace(),
+	}, &ingressList); err != nil {
+		return err
+	}
+	// If particular namespace has only one ingress object then after deletion namespace will have empty ingress object
+	// So remove namespace from SMMR
+	if len(ingressList.Items) == 1 {
```

Comment:
Should we doublecheck if this 1 item is the item we're reconciling currently? I could imagine weird races if two things are deleted in parallel.